README.md

```diff
@@ -28,30 +32,59 @@
 
 Configuration Anomaly Detection (CAD) is responsible for reducing manual SRE effort by pre-investigating alerts, detecting cluster anomalies and sending relevant communications to the cluster owner.
 
-## Contributing
+## Contributing 
 
-To contribute to CAD, please see our [CONTRIBUTING Document](CONTRIBUTING.md).
+### Adding a new investigation
 
-# Documentation
+CAD investigations are triggered by PagerDuty webhooks. Currently, CAD supports the following two formats of webhooks:
+-  WebhookV3 
+-  EventOrchestrationWebhook
 
-## CAD CLI
+The required investigation is identified by CAD based on the incident and its payload. 
+As PagerDuty itself does not provide finer granularity for webhooks than service-based, CAD filters out the alerts it should investigate. For more information, please refer to https://support.pagerduty.com/docs/webhooks.
+
+To add a new alert investigation:
+- create a mapping for the alert to the `getInvestigation` function in `investigate.go` and write a corresponding CAD investigation (e.g. `Investigate()` in `chgm.go`).
+- if the alert is not yet routed to CAD, add a webhook to the service your alert fires on.
+
+## Testing locally
+
+### Pre-requirements
+- an existing cluster
+- an existing PagerDuty incident for the cluster and alert type that is being tested
+
+### Running cadctl for an incident ID
+1) Export the required ENV variables, see [required ENV variables](#required-env-variables).
+2) Create a payload file containing the incident ID
+  ```bash
+  export INCIDENT_ID=
+  echo '{"event": {"data":{"id": "${INCIDENT_ID}"}}}' > ./payload
+  ```
+1) Run `cadctl` using the payload file
+  ```bash
+  ./cadctl/cadctl investigate --payload-path payload
```

Comment:
bonus points if you make step 2 and 3 as a oneliner 
( ah typo as well in the enum ;) )

Reply:
I kept the steps separate to make it clear what is being done. To re-test, you would only have to re-run step 3. Fixed the typo, thanks! :)